heltour/tournament/oauth.py

```diff
@@ -59,23 +60,19 @@ def login_with_code(request, code, encoded_state):
     # oauth_token.account_email = _get_account_email(oauth_token)
     player = Player.get_or_create(username)
 
-    # Ensure the player's profile is present so we can display ratings, etc.
-    _ensure_profile_present(player)
-
     # At this point all http requests are successful, so we can start persisting everything
     oauth_token.save()
     player.oauth_token = oauth_token
     player.save()
 
-    user = User.objects.filter(username__iexact=username).first()
-    if not user:
-        # Create the user with a password no one will ever use; it can always be manually reset if needed
-        with reversion.create_revision():
-            reversion.set_comment('Create user from lichess OAuth2 login')
-            user = User.objects.create_user(username=username, password=create_api_token())
+    # a password starting with ! signifies an unusable password to django
+    user, _ = User.objects.get_or_create(username=normalize('NFKC', username), defaults={"password": f"!{create_api_token(length=40)}"})
```

Comment:
Make the password generation be a function with some notes/documentation (like the comment above).  Same with the username generation. Making them be their own function and putting all of the context discussed into the comments will help future us understand these decisions.

The functions aren't strictly necessary, but they help contextualize the comments. The functions can even be defined inline here, just makes it easier to see that "These comments are about the password generation" and "these comments are about username generation", both of which are orthogonal concerns to the `User.objects.get_or_create(`